libhb/decavcodec.c

```diff
@@ -519,35 +410,65 @@ static int decavcodecaWork( hb_work_object_t * w, hb_buffer_t ** buf_in,
 
     *buf_out = NULL;
 
-    if ( in->s.start < 0 && pv->pts_next <= 0 )
-    {
-        // discard buffers that start before video time 0
-        return HB_WORK_OK;
-    }
-
-    int pos, len;
-    for ( pos = 0; pos < in->size; pos += len )
-    {
-        uint8_t *pout;
-        int pout_len;
-        int64_t cur;
-
-        cur = in->s.start;
+    int     pos, len;
+    int64_t pts = in->s.start;
+
+    // There are a 3 scenarios that can happen here.
+    // 1. The buffer contains exactly one frame of data
+    // 2. The buffer contains multiple frames of data
+    // 3. The buffer contains a partial frame of data
+    //
+    // In scenario 2, we want to be sure that the timestamps are only
+    // applied to the first frame in the buffer.  Additional frames
+    // in the buffer will have their timestamps computed in sync.
+    //
+    // In scenario 3, we need to save the ancillary buffer info of an
+    // unfinished frame so it can be applied when we receive the last
+    // buffer of that frame.
+    if (!pv->unfinished)
+    {
+        // New packet, and no previous data pending
+        pv->packet_info.scr_sequence = in->s.scr_sequence;
+        pv->packet_info.new_chap     = in->s.new_chap;
+        pv->packet_info.frametype    = in->s.frametype;
+    }
+    for (pos = 0; pos < in->size; pos += len)
+    {
+        uint8_t * pout;
+        int       pout_len;
+        int64_t   parser_pts;
 
         if ( pv->parser != NULL )
         {
-            len = av_parser_parse2( pv->parser, pv->context, &pout, &pout_len,
-                                in->data + pos, in->size - pos, cur, cur, 0 );
-            cur = pv->parser->pts;
+            len = av_parser_parse2(pv->parser, pv->context, &pout, &pout_len,
+                                   in->data + pos, in->size - pos,
+                                   pts, pts, 0 );
+            parser_pts = pv->parser->pts;
+            pts = AV_NOPTS_VALUE;
```

Comment:
Not needed in this case since pts is int64_t.  It's only needed when the variable is a float or double.